packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -825,108 +1005,98 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
     return result;
   }
 
-  @override
-  bool get sizedByParent => true;
-
   @override
   Size computeDryLayout(BoxConstraints constraints) {
     assert(constraints.biggest.isFinite);
     return constraints.biggest;
   }
 
   @override
-  void performLayout() {
-    _hasVisualOverflow = false;
-
-    if (_onstageChildCount == 0) {
-      return;
-    }
-
-    _resolve();
-    assert(_resolvedAlignment != null);
-
-    // Same BoxConstraints as used by RenderStack for StackFit.expand.
-    final BoxConstraints nonPositionedConstraints = BoxConstraints.tight(constraints.biggest);
-
+  // The following uses sync* because concurrent modifications should be allowed
+  // during layout.
```

Comment:
This isn't a good reason to use a generator function.

Your callers will _always_ materialize this entire list. Just do it for them and return the list. Then they can modify what they need and not get surprized if calling `length` multiple times is expensive or that the code size has ballooned.

Reply:
(also, I have not read this whole patch and do not expect to - there are already a bunch of cooks in here!)

Reply:
The benchmark Jonah showed me seems to indicate the performance of `sync*` has improved by a lot: https://flutter-flutter-perf.skia.org/e/?queries=sub_result%3DcombineSemanticsInfoList_iteration%26sub_result%3DcombineSemanticsInfoSyncStar_iteration&selected=commit%3D31839%26name%3D%252Carch%253Dintel%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253DMoto_G__4_%252Cdevice_version%253Dnone%252Chost_type%253Dmac%252Csub_result%253DcombineSemanticsInfoSyncStar_iteration%252Ctest%253Dmicrobenchmarks%252C

Reply:
I did a quick benchmark and that also showed time is almost on par with creating a list and iterate over that. The binary size of the two version of app are the same in terms of blocks but I'm not sure that means anything. What's the best way to benchmark the binary size?

Reply:
(posting the benchmark here)

Reply:
It would probably be worht adding to the comment here that the results of iteration can change during the course of iteration because layout will get called between iterator moves and may change what the next item is or whether there is a next item.

Reply:
After switching to release builds there is an 822-byte difference in binary size between the 2 versions of the benchmark app.